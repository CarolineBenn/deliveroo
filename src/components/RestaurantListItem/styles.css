.restaurantListItem__container {
  background: #FFFFFF;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem;
  border-bottom: 1px solid #E5E5E5;
}

.restaurantListItem__name {
  font-size: 18px;
}

.restaurantListItem__details {
  display: flex;
  flex-direction: column-reverse;
  font-size: 14px;
  text-align: right;
  color: #9B9B9B;
}

.restaurantListItem__detail {
  line-height: normal;
}

/**
 * TODO.
 * THINK OF BETTER NAME FOR 'ROW';
 */
 .restaurantListItem__row {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.restaurantListItem__arrow {
  padding-left: 1rem;
}