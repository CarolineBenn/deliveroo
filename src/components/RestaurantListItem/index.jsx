import React from 'react';

import Arrow from '../Arrow';

import './styles.css';

const RestaurantListItem = ({
  id,
  name,
  tags,
  priceCategory,
  time,
  permalink,
}) => (
<li className="restaurantListItem__container">
  <h3 className="restaurantListItem__name">{name}</h3>
  <div className="restaurantListItem__row">
    <div className="restaurantListItem__details">
      <span className="restaurantListItem__detail">{`${time} minutes`}</span>
      <span className="hideBelowMDInline">{priceCategory}</span>
      <span className="restaurantListItem__detail">{tags && tags.join('ãƒ»')}</span>
    </div>
    <div><Arrow fill="#1EBAB2" /></div>
  </div>
</li>
);

export default RestaurantListItem;
