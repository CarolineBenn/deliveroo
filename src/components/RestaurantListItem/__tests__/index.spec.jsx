import React from 'react';
import renderer from 'react-test-renderer';

import RestaurantListItem from '..';

// Mock components that have their own snapshot tests
jest.mock('../../Arrow', () => 'Arrow');

test('RestaurantListItem renders correctly', () => {
  const tree = renderer.create(
    <RestaurantListItem />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});