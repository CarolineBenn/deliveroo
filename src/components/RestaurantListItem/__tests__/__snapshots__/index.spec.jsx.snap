// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantListItem renders correctly 1`] = `
<li
  className="restaurantListItem__container"
>
  <h3
    className="restaurantListItem__name"
  >
    Bobo Social
  </h3>
  <div
    className="restaurantListItem__row"
  >
    <div
      className="restaurantListItem__details"
    >
      <span
        className="restaurantListItem__detail"
      >
        30 minutes
      </span>
      <span
        className="hideBelowMDInline"
      >
        ££
      </span>
      <span
        className="restaurantListItem__detail"
      >
        Vietnamese
      </span>
    </div>
    <div>
      <Arrow
        fill="#1EBAB2"
      />
    </div>
  </div>
</li>
`;
